src/components/SlateEditor/RichTextEditor.tsx

```diff
@@ -220,6 +226,66 @@ const RichTextEditor = ({
 
   const handleKeyDown = useCallback(
     (e: KeyboardEvent<HTMLDivElement>) => {
+      const getElementByType = (parent: HTMLElement, type: string): HTMLElement | null => {
+        if (parent.children.length > 0) {
+          for (let i = 0; i < parent.children.length; i++) {
+            const child = parent.children[i] as HTMLElement;
+            if (child.tagName.toLowerCase() === type) {
+              return child;
+            }
+            if (child.children.length > 0) {
+              const nextChild = getElementByType(child, type);
+              if (nextChild) {
+                return nextChild;
+              }
+            }
+          }
+        }
+        return null;
+      };
+
+      const selectedElement = getCurrentBlock(editor, TYPE_PARAGRAPH) || getCurrentBlock(editor, TYPE_HEADING) || [];
+      const [listBlock] = getCurrentBlock(editor, TYPE_DEFINITION_LIST) || getCurrentBlock(editor, TYPE_LIST) || [];
+
+      if (e.key === KEY_TAB && selectedElement && selectedElement.length > 0 && !listBlock) {
+        const path = ReactEditor.findPath(editor, selectedElement[0]!);
+        if (!e.shiftKey && !Editor.after(editor, path)) return; // If there is no block after the current block, and shift is not pressed, move out from the editor
+        if (e.shiftKey && !Editor.before(editor, path)) return; // If there is no block before the current block and shift is pressed, move out from the editor
+
+        let target = e.target as HTMLElement;
+        if (target.parentNode instanceof HTMLElement) {
+          let nextElement = target.parentNode.nextElementSibling as HTMLElement | null;
+
+          while (!nextElement) {
+            // Keeps looking until it finds the next focusable element
+            target = target.parentNode as HTMLElement;
+            if (target.parentNode instanceof HTMLElement) {
+              const el = target.parentNode.nextElementSibling as HTMLElement | null;
+              if (el) {
+                const button = getElementByType(el, "button");
+                button?.focus();
+                nextElement = button === document.activeElement ? el : null;
+              }
+            }
+          }
+        }
+
+        // let nodeToMoveTo: Descendant[] | Node | null = null;
+        // nodeToMoveTo = getNextNode(editor, path, e.shiftKey);
+
+        // if ("type" in nodeToMoveTo) {
+        //   const element = getElementByType(e.target as HTMLElement, "button");
+        //   // element?.focus();
+        //   return;
+        //   e.preventDefault();
+        //   // if (Editor.isVoid(editor, nodeToMoveTo)) Transforms.move(editor, { unit: "offset" });
+        //   // Transforms.select(editor, ReactEditor.findPath(editor, nodeToMoveTo));
+        //   // Transforms.collapse(editor);
+        //   // ReactEditor.focus(editor);
+        //   // return;
+        // }
```

Comment:
Regner med dette skal slettes.

Reply:
Ja; det er utkommentert kode som er nødvendig for den "duplikate" løsningen til piltastene. Dersom dette fungerer funksjonelt slik vi vil det skal være, vil dette fjernes sammen med noe utilitykode som ble lagt inn med samme formål, men som ikke lenger vil være nødvendig.